File: Data Structures & Algorithms/minimum-size-subarray-sum/submission-3.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        
        int n = nums.length;
        int maxLen = Integer.MAX_VALUE;
        int l = 0, r = 0;
        int sum = 0;

        while( r < n)
        {
            sum += nums[r];

            while( l <= r && sum >= target)
            {
                int len = r - l + 1;
                maxLen = Math.min(len,maxLen);
                sum -= nums[l];
                l++;
            }

            r++;
        }

        return maxLen == Integer.MAX_VALUE ? 0 : maxLen;
    }
}

File: Data Structures & Algorithms/two-integer-sum/submission-3.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        _map = {}
        n = len(nums)
        ans = [] * 2

        for i in range(n):
            if target - nums[i] in _map:
                ans.append(_map[target-nums[i]])
                ans.append(i)
                break
            else:
                _map[nums[i]] = i
        
        return ans